html/user/account_ownership.php

```diff
@@ -0,0 +1,186 @@
+      $sig_verification = openssl_verify($message_data, $decoded_sig, $pubkeyid, OPENSSL_ALGO_SHA512); // Verify that the generated signature against the original data, using the public key.
+      openssl_free_key($pubkeyid); // Free the public key from memory
+
+      // Check if signature was successfully validated
+      if ($sig_verification == 1) {
+          $url_tokens = url_tokens($user->authenticator);
+          // The generated signature has been successfully verified using the public key.
+          global $master_url; // Define global master_url variable for use in output
+          echo "<p>Do not share this information with anyone other than the external system which has requested this proof of account ownership.</p>";
+          echo "<textarea rows='13' cols='50' id='result_textbox'><account_ownership_verification>\n<master_url>$master_url</master_url>\n<msg>$message_data</msg>\n<signature>$base64_sig</signature>\n</account_ownership_verification></textarea>";
+          echo "<br/><br/><button class='btn btn-success' onclick='copy_result_textbox()'>Copy text</button>";
+          echo "<a href='account_ownership.php?$url_tokens'><button class='btn btn-default'>Go back</button></a>";
+          echo '<script type="text/javascript">';
+          echo 'function copy_result_textbox() {
+                var target_textbox = document.getElementById("result_textbox");
+                target_textbox.select();
+                document.execCommand("copy");
+                alert("Copied to clipboard");
+              }';
+          echo '</script>';
+          page_tail();
+
+      } elseif ($sig_verification == 0) {
+          // The generated signature has not been verified. The private/public keys do not match.
+          error_page(tra("Signature verification failed. Contact the project administrator to resolve the issue."));
+      } else {
+          // Something has gone wrong & an error has occurred.
+          error_page(tra("An error occured during the signature verification. Contact the project administrator to resolve the issue."));
+      }
+  } else {
+      // User data input invalid
+      error_page(tra("Invalid input. User input must have a length > 0 and < 4096. <form><input type='button' value='Go back!'' onclick='history.back()'></form>"));
+  }
+}
+
+function account_ownership_form($user) {
+  // GET request - the user has navigated to the page.
+
+  page_head(tra("Generate proof of account ownership"), null, null, null, boinc_recaptcha_get_head_extra());
+
+  if ($user) { // Verify the user is logged in
+      // If the following keys do not exist, then the users will be shown an error message.
+      $config = get_config();
+      $keydir = parse_config($config, "<key_dir>");
+      $private_key_path = "file://$keydir/account_ownership_private.pem";
+      $public_key_path = "file://$keydir/account_ownership_public.pem";
```

Comment:
These four lines are repeated. As a result, this logic should be moved into an include file (maybe create inc/account_ownership.inc?)